### DISCUSSION TITLE
Calculating SPO2 and Heartrate

### DISCUSSION THREAD

**Discussion by IamMugiChan** [n-elia/MAX30102-MicroPython-driver#22] [Help]
Hi there 

I have successfully run the sample file however is it possible to compute 

Heart Rate and SpO2 using this library?

**Comment by n-elia** [comment]
Hi,

this library has the only scope of providing you an access to the MAX30102 sensor when using MicroPython. Therefore, you can only get the RAW data.

To compute HR and SpO2, you have to code your own application. I left some hints about that [here](https://github.com/n-elia/MAX30102-MicroPython-driver#heartrate-and-spo2-estimation). There are some C++ implementations that could be ported in Python.

Feel free to start a new [discussion](https://github.com/n-elia/MAX30102-MicroPython-driver/discussions) if you want to ask the community or to share your findings (greatly appreciated). Also, you may take a look or start a new thread on [MicroPyhton forum](https://forum.micropython.org/).

Bye,
Nicola

**Comment by IamMugiChan** [comment]
Device Used: mh-et live max30102
Library Used: https://pypi.org/project/micropython-max30102/

Reading Sample No Finger Input:
`
- 130 , 114
- 129 , 117
- 128 , 114
`
Reading Sample With Finger Input

- 16338 , 12171
- 15992 , 12597
- 16093 , 12496
- 15879 , 12468
- 16195 , 12467

Now we can say that the device is reading ...

module configuration:
`sensor.set_sample_rate(400)`
`sensor.set_fifo_average(8)`
`sensor.set_active_leds_amplitude(MAX30105_PULSE_AMP_MEDIUM)`

given this data from function
`red_reading = sensor.pop_red_from_storage()`
`ir_reading = sensor.pop_ir_from_storage()`


how can we compute the SpO2 and the Heart Rate? any ideas or sample?

**Comment by n-elia** [comment]
Hi, I've just added a heart rate monitoring example to our MAX30102 MicroPython driver repo!
This simple demo shows how to use the sensor for on-the-fly heart rate (BMP) estimation on devices like the ESP32.
Take a look at it at [`examples/heart_rate`](https://github.com/n-elia/MAX30102-MicroPython-driver/tree/235057b4f47c7c6b568f35a02046c0f394245a08/examples/heart_rate).
